src/task/ReceiveVideoStreamIndexTask.ts

```diff
@@ -220,25 +220,23 @@ export default class ReceiveVideoStreamIndexTask
 
     // Intersect `this.context.videoSendCodecPreferences` with `index.supportedReceiveCodecIntersection`
     for (const capability of this.context.videoSendCodecPreferences) {
-      let codecSupported = false;
       for (const signaledCapability of index.supportedReceiveCodecIntersection) {
         if (capability.equals(VideoCodecCapability.fromSignaled(signaledCapability))) {
-          codecSupported = true;
           newMeetingSupportedVideoSendCodecPreferences.push(capability);
           break;
         }
       }
-      // We need to renegotiate if we are currently sending a codec that is no longer supported in the call.
+    }
+
+    if (newMeetingSupportedVideoSendCodecPreferences.length > 0) {
+      // We need to renegotiate if we are currently sending a codec that is no longer supported in the call, or if we can
+      // start sending a higher preference codec again (due to a constrained remote attendee leaving)
       if (
         this.context.currentVideoSendCodec !== undefined &&
-        !codecSupported &&
-        capability.equals(this.context.currentVideoSendCodec)
+        !newMeetingSupportedVideoSendCodecPreferences[0].equals(this.context.currentVideoSendCodec)
```

Comment:
So the first codec in videoSendCodecPreferences is always the current send codec?

Reply:
I thought so...but now that I think about it there is a pretty unlikely edge case where the preference list starts with a codec that can be received by all clients (including the sender) but not sent. While this is not possible on any existing browser/codec, let me see if I can harden this.